test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1

```diff
@@ -4,31 +4,47 @@ Describe "Job Cmdlet Tests" -Tag "CI" {
             Get-Job | Remove-Job -force
         }
         BeforeEach {
-            $j = start-job -scriptblock { 1 + 1 }
+            $j = start-job -scriptblock { 1 + 1 } -Name "My Job"
         }
         It "Start-Job produces a job object" {
-            $j.gettype().fullname | should be "System.Management.Automation.PSRemotingJob"
+            $j | should beoftype System.Management.Automation.Job
+            $j.Name | should be "My Job"
         }
         It "Get-Job retrieves a job object" {
-            (Get-Job -id $j.id).gettype().fullname | should be "System.Management.Automation.PSRemotingJob"
+            (Get-Job -id $j.id) | should beoftype System.Management.Automation.Job
+        }
+        It "Get-Job retrieves an array of job objects" {
+            start-job -ScriptBlock { 2 * 2 }
+            $jobs = Get-Job
+            $jobs.Count | should be 2
+            foreach ($job in $jobs) {
+                $job | should beoftype System.Management.Automation.Job
+            }
         }
         It "Remove-Job can remove a job" {
             remove-job $j -force
             try {
-                get-job $j -ea Stop
+                get-job $j -ErrorAction Stop
                 throw "Execution OK"
             }
             catch {
                 $_.FullyQualifiedErrorId | should be "JobWithSpecifiedNameNotFound,Microsoft.PowerShell.Commands.GetJobCommand"
             }
         }
-        It "Receive-Job can retrieve job results" -pending {
+        It "Receive-Job can retrieve job results" {
             $waitjob = Wait-Job -Timeout 10 -id $j.id
             if ( $waitJob ) {
                 $result = receive-job -id $j.id
-                $result.id | Should be 2
+                $result | Should be 2
             }
```

Comment:
Next test implicitly check the same but it can be modified in future so we should leave the test with corrections:
```powershell
        It "Receive-Job can retrieve job results" {
            Wait-Job -Timeout 10 -id $j.id | Should Not BeNullOrEmpty
            receive-job -id $j.id | Should be 2
        }
```